lightning/src/onion_message/messenger.rs

```diff
@@ -267,14 +267,27 @@ impl Responder {
 		}
 	}
 
-	/// Creates the appropriate [`ResponseInstruction`] for a given response.
+	/// Creates a [`ResponseInstruction::WithoutReplyPath`] for a given response.
+	///
+	/// Use when the recipient doesn't need to send back a reply to us.
 	pub fn respond<T: OnionMessageContents>(self, response: T) -> ResponseInstruction<T> {
 		ResponseInstruction::WithoutReplyPath(OnionMessageResponse {
 			message: response,
 			reply_path: self.reply_path,
 			path_id: self.path_id,
 		})
 	}
+
+	/// Creates a [`ResponseInstruction::WithReplyPath`] for a given response.
+	///
+	/// Use when the recipient needs to send back a reply to us.
+	pub fn respond_with_reply_path<T: OnionMessageContents>(self, response: T) -> ResponseInstruction<T> {
+		ResponseInstruction::WithReplyPath(OnionMessageResponse {
+			message: response,
+			reply_path: self.reply_path,
+			path_id: self.path_id,
+		})
+	}
```

Comment:
Could you add a test for this behavior?

Reply:
Introduce a new test in [27899b1](https://github.com/lightningdevkit/rust-lightning/commit/27899b1a04bfd78db34f2c22ad489f1a6fd4e5e4) to test this behavior!
Thanks!